mockup/patterns/structure/js/views/textfilter.js

```diff
@@ -93,7 +93,7 @@ define([
         clearTimeout(self.timeoutId);
       }
       self.timeoutId = setTimeout(function() {
-        self.term = $(event.currentTarget).val();
+        self.term = encodeURIComponent($(event.currentTarget).val());
```

Comment:
this turns spaces into %20, for instance, and is more used to encode get params. does the filter decode already?

alternatively we could transform 'Text1 & Text2' into `Text1 &amp; Text2` with `new Option($(event.currentTarget).val()).innerHTML` _(maybe `event.currentTarget.value`, without jquery, works even in ie11)_. but still the filter would need to decode these.

Reply:
@davilima6 I honestly don't mind any solution... I used encodeURIComponent because it works and encodes everything properly... For instance, I submit the following text in the filter query:

`This is my text including special characters , / ? : @ & = + $ # and some ét®æß`

I then see this under the network tab 
![image](https://user-images.githubusercontent.com/851006/71724379-e6833600-2e0e-11ea-99ab-61f7b93c2926.png)

And I see this in the backend
```
(Pdb) query
{'criteria': [{'i': 'SearchableText', 'o': 'plone.app.querystring.operation.string.contains', 'v': 'This is my text including special characters , / ? : @ & = + $ # and some 
ét®æß*'}, {'i': 'path', 'o': 'plone.app.querystring.operation.string.path', 'v': '/::1'}], 'sort_on': 'getObjPositionInParent', 'sort_order': 'ascending'}
```

(This is with a pdb in line 636 of `plone.app.vocabularies.catalog`)

If your solution works like this, then I'm happy :)

Reply:
If the filter already parses encoded URIs then no need for change, thanks for the fix